migra/migra.py

```diff
@@ -75,6 +81,8 @@ def add_extension_changes(self, creates=True, drops=True):
             self.add(self.changes.extensions(drops_only=True))
 
     def add_all_changes(self, privileges=False):
+        formatter = Formatter()
```

Comment:
this isn't even a singleton, everytime the method is called you'll create a new object

storing it as an object attribute or a module variable makes more sense here